src/main/java/Parser.java

```diff
@@ -0,0 +1,123 @@
+                tasks.add(event);
+                System.out.println("Got it, I've added the following task:\n" + "  " + event + "\n"
+                        + "Now you have " + tasks.size() + " tasks in the list.");
+                break;
+            }
+            case "delete": {
+                int toDelete = Integer.parseInt(input.substring(input.indexOf(' ') + 1, input.length())) - 1;
+                Task task = tasks.get(toDelete);
+                tasks.remove(toDelete);
+                System.out.println("Noted, I've removed the following task:\n" + "  " + task + "\n"
+                        + "Now you have " + tasks.size() + " tasks in the list.");
+                break;
+            }
+            case "done": {
+                int toEdit = Integer.parseInt(input.substring(input.indexOf(' ') + 1, input.length())) - 1;
+                Task task = tasks.get(toEdit);
+                task.markAsDone();
+                break;
+            }
+            case "find": {
+                String keyword = input.substring(input.indexOf(' ') + 1);
+                ArrayList<Task> matchingTasks = tasks.findTasks(keyword);
+                System.out.println("Here are the matching tasks in your list: ");
+                for (int i = 0; i < matchingTasks.size(); i++) {
+                    System.out.println(i+1 + "." + matchingTasks.get(i));
+                }
+            }
+        }
+
+    }
+
+
+    /**
+     * Gets the command out from user input
+     * @param input is the user input
+     * @return a String that is only the command
+     * @throws DukeException is thrown when there is an invalid command by user
+     */
+    public String getCommand(String input) throws DukeException {
+        if (!input.contains(" ")) {
+            // check if the command is correct
+            if (!input.equals("todo") &&
+                    !input.equals("deadline") &&
+                    !input.equals("event") &&
+                    !input.equals("delete") &&
+                    !input.equals("done") &&
+                    !input.equals("find")) {
+                throw new DukeException("OOPS! I'm sorry but I dont't know what that means :(");
+            } else {
+                // command is not valid
+                throw new DukeException("OOPS! The description of a " + input + " cannot be empty");
+            }
+        } else {
+            if (!input.substring(0, input.indexOf(' ')).equals("todo") &&
+                    !input.substring(0, input.indexOf(' ')).equals("deadline") &&
+                    !input.substring(0, input.indexOf(' ')).equals("event") &&
+                    !input.substring(0, input.indexOf(' ')).equals("delete") &&
+                    !input.substring(0, input.indexOf(' ')).equals("done") &&
+                    !input.substring(0, input.indexOf(' ')).equals("find")) {
+                throw new DukeException("OOPS! I'm sorry but I dont't know what that means :(");
```

Comment:
It would be neater if a `Command` class is implemented for better abstraction. What happens if more commands are added? It may become more difficult to maintain and keep track of all the individual commands.
Alternatively, you can also have an array/HashSet of commands.